libs/content/workspace/src/storage_viewer/data.rs

```diff
@@ -0,0 +1,162 @@
+                    .size_bytes,
+            });
+        }
+
+        filerows
+    }
+
+    pub fn init(core: Lb, potential_root: Option<File>) -> Self {
+        let data = Self::get_filerows(core);
+        let mut all_files = HashMap::new();
+        let mut root = Uuid::nil();
+        for datum in data.clone() {
+            if datum.file.id == datum.file.parent {
+                root = datum.file.id;
+            }
+            all_files.insert(datum.file.id, datum);
+        }
+
+        let mut folder_sizes = HashMap::new();
+        //Initial for loop for folders is necessary to give folders starting value as we need to go over folders again to update sizes
+        for datum in data.clone() {
+            if datum.file.is_folder() {
+                folder_sizes.insert(datum.file.id, datum.size);
+            }
+        }
+        for datum in data {
+            let datum_size = datum.size;
+            let mut current_id = datum.file.id;
+            loop {
+                let row = all_files.get(&current_id).unwrap();
+                let mut current_size = folder_sizes
+                    .get(&row.file.parent)
+                    .copied()
+                    .unwrap_or_default();
+                current_size += datum_size;
+                if current_id == root {
+                    break;
+                }
+                folder_sizes.insert(row.file.parent, current_size);
+                current_id = row.file.parent;
+            }
+        }
+
+        let folder_root = match potential_root {
+            Some(folder_root) => folder_root.id,
+            None => root,
+        };
+
+        Self { current_root: folder_root, overall_root: root, all_files, folder_sizes }
+    }
+
+    pub fn get_children(&self, id: &Uuid) -> Vec<Node> {
+        if !self.all_files.get(id).unwrap().file.is_folder() {
+            return vec![];
+        }
+        let total_size = *self.folder_sizes.get(&self.current_root).unwrap() as f32;
+        let children = self
+            .all_files
+            .values()
+            .filter(|f| f.file.parent == *id && f.file.id != *id)
+            .map(|f| {
+                let mut current_size = f.size as f32;
+                if f.file.is_folder() {
+                    current_size = *self.folder_sizes.get(&f.file.id).unwrap() as f32;
+                }
+                Node {
+                    id: f.file.id,
+                    name: f.file.name.clone(),
+                    portion: current_size / total_size,
+                    children: self.get_children(&f.file.id),
+                }
+            });
+        let mut gathered_children = vec![];
+        for child in children.into_iter() {
+            gathered_children.push(child);
+        }
+        gathered_children.sort_by(|a, b| {
+            let a_size = (a.portion * 10000.0) as u32;
+            let b_size = (b.portion * 10000.0) as u32;
```

Comment:
something very strange is going on here...

consider just `sort_by_key(|a| a.portion)`

If it's some f32 weirdness going on try this: https://users.rust-lang.org/t/how-to-sort-a-vec-of-floats/2838/2